src/react-access-context.js

```diff
@@ -0,0 +1,43 @@
+import {Component, PropTypes} from 'react';
+import _ from 'lodash';
+
+export default class ReactAccessContext extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    userPermissions: PropTypes.arrayOf(PropTypes.string),
+    validator: PropTypes.func,
+  }
+
+  render() {
+    return this.props.children;
+  }
+
+  getChildContext = () => ({
+    authorizeAccess: this.authorizeAccess
+  })
+
+  authorizeAccess = (requiredPermissions, requireAll) => {
+    const {userPermissions, validator} = this.props;
+    return validator(userPermissions, requiredPermissions, requireAll);
+  }
+
+  static childContextTypes = {
+    authorizeAccess: PropTypes.func,
+  }
+
+  static defaultProps = {
+    validator(userPermissions, requiredPermissions, requireAll) {
+      if (requireAll === true) {
+        // TODO: userPermissions contains all requiredPermissions
+        // return
+      } else {
+        // by default we will authorize access if any of the permissions in
+        // requiredPermissions match any userPermissions
+        // TODO: is there a smaller comparision function we can use to reduce
+        // bundle size and drop a dependency?
+        return _.intersection(userCapabilities, requiredRoles).length > 1;
+      }
+    },
+    userPermissions: [],
+  }
```

Comment:
yea, I was trying out this ordering pattern:

1. propTypes / contextTypes
1. render
1. literallyEverythingIDGAF

My thought was: 

- the top two explain the incoming API and what you might expect to pass it, or be able to use
- render has been at the bottom for me forever, and it's dumb, @tbranyen is right it should be at the top to help explain what the output is quicker
- lifecycle/helpers can be below that if you want to dig in.

What do you think, still weird?